src/content/sidebar/sidebar-wrapper.tsx

```diff
@@ -1,59 +1,136 @@
 import * as browser from "webextension-polyfill"
 
-import {css} from "@emotion/react"
-import {slide as Slider} from 'react-burger-menu'
-import {useEffect, useState} from "react"
+import {useEffect, useState, useRef} from "react"
 import {useTabLocalState} from "../../core/react"
 
 export const SidebarWrapper = () => {
+    const wrapperRef = useRef()
+    const frameRef = useRef()
+
     const [isOpen, setOpen] = useTabLocalState("sidebarOpen", false)
+
+    // the iframe should be rendered as soon as `isOpen` is true but
+    // when `isOpen` becomes `false` the iframe should only dissapear
+    // after the hiding animation is completed 
+    const [showIFrame, setShowIFrame] = useState(isOpen)
+
     // todo show a loading indicator instead of emptiness
     const [url, setUrl] = useState("")
 
+    const [width, setWidth] = useState(15)
+    const [isLeft, setLeft] = useState(false)
+    const [isDragging, setDragging] = useState(false)
+
     useEffect(() => {
         const messageCallback = (event) => {
             console.log('content-script container ev', event)
-            if (event.type === 'update-panel-url') {
-                setUrl(event.url)
-            } else if (event.type === "toggle-sidebar") {
-                setOpen(!isOpen)
+            if ('type' in event) {
+                if (event.type === 'update-panel-url') {
+                    setUrl(event.url)
+                } else if (event.type === "toggle-sidebar") {
+                    setOpen(!isOpen)
+                }
+            }
+        }
+
+        const windowMessageCallback = (event) => {
+            // make sure the message wasn't sent by a different iframe 
+            if (frameRef?.current?.getAttribute('src').startsWith(event.origin)) {
+                if ('data' in event) {
+                    // redirect the relevant part of the message to the actual callback
+                    messageCallback(event.data)
+                }
             }
         }
 
+        if (isOpen) {
+            setShowIFrame(true)
+        } else {
+            wrapperRef.current.addEventListener('transitionend', () => {
+                if (wrapperRef.current.classList.contains('sidebar--closed')) {
+                    setShowIFrame(false)
+                    wrapperRef.current.removeListener('transitionend', this)
+                }
+            })
+            // hide iFrame after 5s in case transitions are disabled or not supported
+            setTimeout(() => {
+                if (wrapperRef.current.classList.contains('sidebar--closed')) {
+                    setShowIFrame(false)
+                }
+            }, 5000);
```

Comment:
why do we need to use Refs for this? 

this seems like it should be implementable as a part of standard rendering flow?

Reply:
I removed the whole part. Since its not necessary to unload the iframe once it was loaded.